## Add workaround for compiler assertion failure in Promise.flatten_next

Under certain conditions a specific usage of `Promise.flatten_next` and `Promise.next` could
trigger a compiler assertion, whose root-cause is still to be investigated. 

This change avoid the code-path triggering the assertion entirely, and is thus only a workaround, not a full solution, but it increases stability of the stdlib and the promises package.
